File: experiments/source/ocaml/ratio_weighted_sum_reassociated.ml
open Loom

let[@loom.entry] ratio_weighted_sum_reassociated (scale : float) (epsilon : float)
    (x : Tensor1.t) (y : Tensor1.t) =
  let contribution xi yi =
    let scaled = xi *. scale in
    scaled /. (epsilon +. yi)
  in
  Tensor1.reduce_sum (Tensor1.map2 contribution x y)

File: experiments/source/python/ratio_weighted_sum_reassociated.py
import loom


@loom.entry
def ratio_weighted_sum_reassociated(scale: float, epsilon: float, x: loom.Tensor1, y: loom.Tensor1) -> float:
    def contribution(xi: float, yi: float) -> float:
        scaled = xi * scale
        return scaled / (epsilon + yi)

    return loom.Tensor1.reduce_sum(loom.Tensor1.map2(contribution, x, y))
